include/verilated_saif_c.h

```diff
@@ -0,0 +1,430 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//=============================================================================
+//
+// Code available from: https://verilator.org
+//
+// Copyright 2001-2025 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//=============================================================================
+///
+/// \file
+/// \brief Verilated tracing in SAIF format header
+///
+/// User wrapper code should use this header when creating SAIF traces.
+///
+//=============================================================================
+
+#ifndef VERILATOR_VERILATED_SAIF_C_H_
+#define VERILATOR_VERILATED_SAIF_C_H_
+
+#include "verilated.h"
+#include "verilated_trace.h"
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+class VerilatedSaifBuffer;
+class VerilatedSaifFile;
+
+//=============================================================================
+// ActivityBit
+
+class ActivityBit final {
+public:
+    // METHODS
+    VL_ATTR_ALWINLINE
+    void aggregateVal(uint64_t dt, bool newVal) {
+        m_transitions += newVal != m_lastVal ? 1 : 0;
+        m_highTime += m_lastVal ? dt : 0;
+        m_lastVal = newVal;
+    }
+
+    // ACCESSORS
+    VL_ATTR_ALWINLINE bool getBitValue() const { return m_lastVal; }
+    VL_ATTR_ALWINLINE uint64_t getHighTime() const { return m_highTime; }
+    VL_ATTR_ALWINLINE uint64_t getToggleCount() const { return m_transitions; }
+
+private:
+    // MEMBERS
+    bool m_lastVal = false;
+    uint64_t m_highTime = 0;
+    size_t m_transitions = 0;
+};
+
+//=============================================================================
+// ActivityVar
+
+class ActivityVar final {
+public:
+    // CONSTRUCTORS
+    ActivityVar(uint32_t width, ActivityBit* bits)
+        : m_bits{bits}
+        , m_width{width} {}
+
+    ActivityVar(ActivityVar&&) = default;
+    ActivityVar& operator=(ActivityVar&&) = default;
+
+    // METHODS
+    VL_ATTR_ALWINLINE void emitBit(uint64_t time, CData newval);
+
+    template <typename DataType>
+    VL_ATTR_ALWINLINE void emitData(uint64_t time, DataType newval, uint32_t bits) {
+        static_assert(std::is_integral<DataType>::value,
+                      "The emitted value must be of integral type");
+
+        uint64_t dt = time - m_lastTime;
```

Comment:
```suggestion
        const uint64_t dt = time - m_lastTime;
```

Check for other missing const, thanks.